fe/fe-core/src/main/java/com/starrocks/analysis/ColumnDef.java

```diff
@@ -209,12 +223,35 @@ public void analyze(boolean isOlap) throws AnalysisException {
             throw new AnalysisException(String.format("Invalid default value for '%s'", name));
         }
 
-        if (defaultValue.isSet && defaultValue.value != null) {
+        if (defaultValue.isSet && defaultValue.expr instanceof LiteralExpr) {
             try {
-                validateDefaultValue(type, defaultValue.value);
+                if (defaultValue.expr instanceof StringLiteral) {
+                    validateDefaultValue(type, ((StringLiteral) defaultValue.expr).getValue());
+                }
             } catch (AnalysisException e) {
                 throw new AnalysisException(String.format("Invalid default value for '%s': %s", name, e.getMessage()));
             }
+        } else if (defaultValue.isSet) {
+            try {
+                validateDefaultExpr(type, defaultValue.expr);
+            } catch (AnalysisException e) {
+                throw new AnalysisException(String.format("Invalid default expr for '%s': %s", name, e.getMessage()));
+            }
+        }
+    }
+
+    public static void validateDefaultExpr(Type type, Expr defaultExpr) throws AnalysisException  {
+        if (!(defaultExpr instanceof FunctionCallExpr)) {
```

Comment:
It also can be Literal Expr.

Reply:
Literal Expr already check in validateDefaultValue from old code and NullLiteral is not need to check.
you prefer to unify validateDefaultValue and validateDefaultExpr?

Reply:
Yes, I think there is no need to have two functions.

Reply:
ok